l10n_br_fiscal/models/res_company.py

```diff
@@ -41,10 +42,34 @@
 class ResCompany(models.Model):
     _inherit = "res.company"
 
+    sn_effective_tax_ids = fields.One2many(
+        comodel_name="l10n_br_fiscal.simplified.tax.effective",
+        inverse_name="company_id",
+        string="Effective Taxs",
+        help="Simples Nacional effective tax rate for each annex,"
+        "based on the range the company currently falls into.",
+    )
+
+    icmssn_credit_industry = fields.Float(
+        string="ICMS Credit rate for Industry",
+        help="Rate referring to the icms credit allowed when it is a sale of"
+        "industrialized product in the establishment."
+        "Applicable to Simple Nacional.",
+        readonly=True,
+        compute="_compute_icmssn",
+    )
+
+    icmssn_credit_commerce = fields.Float(
+        string="ICMS Credit rate for Commerce",
+        help="Rate referring to the icms credit allowed when it is a resale"
+        "of merchandise. Applicable to Simple Nacional.",
+        readonly=True,
+        compute="_compute_icmssn",
+    )
+
     def _get_company_address_field_names(self):
         partner_fields = super()._get_company_address_field_names()
         return partner_fields + [
-            "tax_framework",
             "cnae_main_id",
         ]
```

Comment:
A ideia desse override era que se vc cria uma nota fiscal para um contato ou endereço de uma certa empresa, esse contato ou endereço vai "herdar" o tax_framework da empresa do qual ele pertence. Vi que agora vc introduziu um _compute_tax_framework, mas esse sistema de herdar o tax_framework da empresa parente vai continuar funcionando se vc tirar o "tax_framework" dessa lista de company_address_field_names?

Reply:
huum, eu penso que quando precisar recuperar as informoções fiscais da empresa do contato ou endereço, não seria melhor acessar a partir do  `partner_id.commercial_partner_id ? até porque tem bem mais campos que estes que podem ser importante para a not fiscal. Isso leva de volta a discussão desta PR:  https://github.com/OCA/l10n-brazil/pull/2520

Mas sou sincero em dizer que não me recordo de qual era o problema com o `tax_framework` que me levou a fazer um compute isolado para ele, vou fazer uma analise melhor e em breve volto aqui com o resultado.

Reply:
talvez vc tirou porque conflitava com o compute, sera?

é valido esse seu questionamento. Sei que a Odoo SA inventou esse lance de ficar copiando alguns campos entre endereços, contatos e empresa parente e que de fato em algumas situações a gente acha mais seguro usar o commercial_partner_id. Vou dar uma pensadinha, mas eu diria que se tiver como deixar o tax_framework sincronizado seria uma certa segurança já...

Reply:
@rvalyi verifiquei aqui e o `_get_company_address_field_names` realmente é apenas para sincronizar os dados entre o `res_company` e o `res_partner` mas isso eu continuo fazendo no método compute e inverse, acho que eu só fiz separado para que ele possa ser computado isoladamente, se não ele computa todo o endereço e os outros campos denovo sem nececidade